tests/utils.py

```diff
@@ -61,3 +63,8 @@ def generate_attribute_map(obj: Union[Product, ProductVariant]) -> Dict[int, Set
         assignment.attribute.pk: {value.pk for value in assignment.values.all()}
         for assignment in qs
     }
+
+
+def clear_temporary_dir(upload_dir):
+    temp_dir = f"{tempfile.gettempdir()}/{upload_dir}"
+    shutil.rmtree(temp_dir)
```

Comment:
There is a fixture for such case. Here is an example of mine on how to use it: https://github.com/NyanKiyoshi/pytest-django-queries/blob/47a1325c811de4f3d5cf0c032b36cd5994e1929e/tests/test_plugin.py#L44-L64

Reply:
It seems that I still need te override `MEDIA_ROOT`. The problem here is that `save_csv_file_in_job` saves CSV file to job and without proper set of `MEDIA_ROOT` test failed.  
Current solution creates new file every time test is running, so because of that cleaning tempdir is necessary. But I can use `tmpdir`as a fixture, override `settings.MEDIA_ROOT` in test and in this case every time new `temp_directory` will be creating, but only 3 last directory are kept. However it will still creats something on your disk and not delete it.

We can also mock save to Django Storage but in this case we can't check if csv file is saved correctly.

Reply:
If I understand correctly, your issue/worry is that pytest would not delete the temporary files? I looked into (was too lazy to try locally 😉 ), and found this:

> Temporary directories are by default created as sub-directories of the system temporary directory. The base name will be pytest-NUM where NUM will be incremented with each test run. Moreover,**entries older than 3 temporary directories will be removed.** - https://docs.pytest.org/en/latest/tmpdir.html#base-temporary-directory

Does it help?

Reply:
Yes, that's true. But it's not so easy case.
I want to test if the proper data in proper format are saved in csv file. So I provide data and file name to function, which creates temporary file and saved it into the job instace. Then I read the file from job instance and checked it's content. 
So I can't use file as a fixture because it's generated in the function. With overriding `MEDIA_ROOT` I ensure that this file will be saved in specific directory (what's more it doesn't work without that ). So it would be best if the directory with this new files will be removed.  

Maybe not so bad is using `tmpdir` as `MEDIA_ROOT` which kept only 3 last directories (https://docs.pytest.org/en/latest/tmpdir.html#the-default-base-temporary-directory) if we don't want to use current solution.

Reply:
I don't understand why you don't want to set `MEDIA_ROOT` to the `str()` of the path given by pytest's fixture?